tests~/MultiDictionaryTests.cs

```diff
@@ -0,0 +1,239 @@
+using System.Diagnostics;
+using CsCheck;
+using SpacetimeDB;
+using Xunit;
+
+public class MultiDictionaryTests
+{
+    /// <summary>
+    /// Generate a list of KeyValuePairs.
+    /// If any two items of the list have duplicate Keys, they are guaranteed to have duplicate Values.
+    /// </summary>
+    /// <typeparam name="TKey"></typeparam>
+    /// <typeparam name="TValue"></typeparam>
+    /// <param name="g1"></param>
+    /// <param name="g2"></param>
+    /// <param name="equality"></param>
+    /// <returns></returns>
+    Gen<List<KeyValuePair<TKey, TValue>>> ListWithNormalizedDuplicates<TKey, TValue>(Gen<TKey> g1, Gen<TValue> g2, IEqualityComparer<TKey> equality, int ListMinLength = 0, int ListMaxLength = 32)
+    where TKey : notnull
+    {
+        return Gen.Select(g1, g2, (b1, b2) => new KeyValuePair<TKey, TValue>(b1, b2)).List[ListMinLength, ListMaxLength].Select(list =>
+        {
+            Dictionary<TKey, TValue> seenKeys = new(equality);
+            for (var i = 0; i < list.Count; i++)
+            {
+                var (b1, b2) = list[i];
+                if (seenKeys.ContainsKey(b1))
+                {
+                    list[i] = new(b1, seenKeys[b1]);
+                }
+                else
+                {
+                    seenKeys[b1] = b2;
+                }
+            }
+            return list;
+        });
+    }
+
+    [Fact]
+    public void Equality()
+    {
+        ListWithNormalizedDuplicates(Gen.Byte[1, 10], Gen.Byte[1, 10], EqualityComparer<byte>.Default).Sample(list =>
+        {
+            var m1 = MultiDictionary<byte, byte>.FromEnumerable(list, EqualityComparer<byte>.Default, EqualityComparer<byte>.Default);
+            Gen.Shuffle(list);
+            var m2 = MultiDictionary<byte, byte>.FromEnumerable(list, EqualityComparer<byte>.Default, EqualityComparer<byte>.Default);
+
+            Assert.Equal(m1, m2);
```

Comment:
The code inside `Sample` is what actually matters in these tests. Basically, `list` is a list of `KeyValuePair<byte, byte>` that is randomly generated, and we test that no matter what value it has, this code works.